presto-docs/src/main/sphinx/functions/aggregate.rst

```diff
@@ -355,3 +360,58 @@ Lambda Aggregate Functions
         -- (2, 42)
 
     The state type must be a boolean, integer, floating-point, or date/time/interval.
+
+
+Filtering During Aggregation
+----------------------------
+
+The ``FILTER`` keyword can be used to remove rows from aggregation processing
+with a condition expressed using a ``WHERE`` clause. This is evaluated for each
+row before it is used in the aggregation and is supported for all aggregate
+functions.
+
+    AGGREATE_FUNCTION(<expression>) FILTER(WHERE <condition>)
+
+For example, you want to add a condition on the count for Iris flowers,
+modifying the following query.
+
+.. code-block:: none
+
+    SELECT species,
+           count(*) AS count
+    FROM memory.default.iris
+    GROUP BY species;
+    species   |  count
+    ------------+-------
+    setosa     |    50
+    virginica  |    50
+    versicolor |    50
+
+If you just use a normal WHERE statement you loose information:
+
+.. code-block:: none
+
+    SELECT species,
+        count(*) AS count
+    FROM memory.default.iris
+    WHERE petallengthcm > 4
+    GROUP BY species;
+    species   | _col1
+    ------------+-------
+    virginica  |    50
+    versicolor |    34
+
+Using a filter you retain all information:
+
+.. code-block:: none
+
+    SELECT species,
+           count(*) FILTER (where petallengthcm > 4) AS count
+    FROM memory.default.iris
+    GROUP BY species;
+    species   | count
```

Comment:
i'd live an empty line between query and data output.
in fact, we sometimes render this as 2 blocks.

example from `accumulo.rst`:

```
.. code-block:: sql

    DESCRIBE myschema.scientists;

.. code-block:: none

      Column   |  Type   | Extra |                      Comment
    -----------+---------+-------+---------------------------------------------------
     recordkey | varchar |       | Accumulo row ID
     name      | varchar |       | Accumulo column name:name. Indexed: false
     age       | bigint  |       | Accumulo column age:age. Indexed: false
     birthday  | date    |       | Accumulo column birthday:birthday. Indexed: false
```